### DISCUSSION TITLE
React production build not working

### DISCUSSION THREAD

**Discussion by CWSites** [parcel-bundler/parcel#8119] [Q&A]
I've got a super basic React application which has HTML, SCSS and JS. When running it locally everything is fine and works perfectly. I've run `yarn build` which doesn't have any errors and the `dist` directory is properly (to the best of my knowledge) compiled. However when I try to open the `dist/index.html` file either directly from my computer OR uploading to a server and viewing it there I get the following errors.

```
GET https://mydomain.com/index.38ec9247.css net::ERR_ABORTED 404
GET https://mydomain.com/dist/App.jsx net::ERR_ABORTED 404
GET https://mydomain.com/index.975ef6c8.js net::ERR_ABORTED 404
```

On another site using parcel I get the following. This site is built using Web 3, you'll need to use Brave Browser or the [unstoppable extension for Chrome/Firefox](https://unstoppabledomains.com/extension) and can be accessed here: [Web 3 Sandbox App](https://gateway.pinata.cloud/ipfs/QmRV9WvBtLxWFFu9B7wz5tW7S9eHNJ3WG17BAv8ac9j3nq/)

```
help.html:1 Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.
help.html:1 Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.
```

Any help or direction would be greatly appreciated. I love how simplistic Parcel is over Create React App but if I can get this to work then there is no point in using it.

## Modifying `dist` files after build

By doing the following I was able to make some progress however I don't understand why Parcel doesn't handle this for me during the build.

- ~I updated anywhere `<script type="module">` to `<script type="text/javascript">`~
- I updated all references to `.js` or `.css` files inside of `index.html` to include `./` at the beginning instead of just `/`

### Source `index.html`
```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>Meta Circus</title>
    <link rel="stylesheet" href="./styles/style.scss" />
    <script type="module" src="./App.jsx"></script>
  </head>
  <body>
    <div id="app"></div>
    <script type="module" src="./index.js"></script>
  </body>
</html>

```

### Build `dist/index.html`
```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>My Website</title>
    <link rel="stylesheet" href="/index.1c2d55b2.css" />
    <script type="module" src="/index.f8289b62.js"></script>
  </head>
  <body>
    <noscript>JavaScript must be enabled to run this app.</noscript>
    <div id="app"></div>
    <script type="module" src="/index.15767dda.js"></script>
  </body>
</html>
```

### Modified `dist/index.html` that works
```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>My Website</title>
    <link rel="stylesheet" href="./index.1c2d55b2.css" />
    <script type="module" src="./index.f8289b62.js"></script>
  </head>
  <body>
    <noscript>JavaScript must be enabled to run this app.</noscript>
    <div id="app"></div>
    <script type="module" src="./index.15767dda.js"></script>
  </body>
</html>

```

**Comment by devongovett** [comment]
I guess your server is serving the files with the text/html mimetype? You need to configure that so it serves text/javascript.

**Reply by CWSites** [reply]
I uploaded the `/dist` directory and I'm accessing the `index.html` file directly. I haven't adjusted anything. One of these is being served through GitHub Pages and the other is through Pinata.cloud (web3 hosting).

I was just able to verify that by adding the `./` in front of the files that I'm importing on the `dist/index.html` it's working on GitHub Pages.

Any ideas there?

**Reply by CWSites** [reply]
Hey @devongovett I've got it working but every time I have to add the `./` in front of the imports for my JS & CSS files. My `src/index.html` has the `./` but it's getting stripped out when built?

**Reply by devongovett** [reply]
I'm not sure why that would be necessary. `<script src="./file.js">` and `<script src="file.js">` are equivalent - the `./` is not necessary because `file.js` is already a relative URL.

**Reply by CWSites** [reply]
I agree and I'm not sure either, but both GitHub Pages and Pinata (web3) don't work without the `./` in front. Is there a way that we can update the build process so that it doesn't alter this output?

**Reply by devongovett** [reply]
You could possibly use `--public-url` to make the URLs absolute: https://parceljs.org/features/targets/#publicurl

**Reply by CWSites** [reply]
Thanks I'll take a look

**Reply by CWSites** [reply]
`--public-url` worked, thank you!
